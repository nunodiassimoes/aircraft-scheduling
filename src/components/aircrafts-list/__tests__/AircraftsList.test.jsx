import "@testing-library/jest-dom/extend-expect";
import { render, waitFor } from "@testing-library/react";

import { aircraftsFixture } from "utils/tests/fixtures";
import mockedAxios from "utils/tests/mockedAxios";

import AircraftsList from "..";

const { mockAxios } = mockedAxios;

const renderAircraftsList = () => {
  return render(<AircraftsList />);
};

describe("AircraftsList", () => {
  beforeEach(() => {
    jest.clearAllMocks();
    mockedAxios.resetAllMocks();
    mockedAxios.get.success(aircraftsFixture);
  });

  test("should display correctly the list of aircrafts", async () => {
    const { container } = renderAircraftsList();

    await waitFor(() => expect(mockAxios.get).toHaveBeenCalledTimes(1));
    expect(container).toMatchSnapshot();
  });
});
