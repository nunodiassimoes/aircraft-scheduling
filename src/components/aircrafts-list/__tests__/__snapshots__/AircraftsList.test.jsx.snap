// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AircraftsList should display correctly the list of aircrafts 1`] = `
<div>
  <h2>
    We have 
    1
     aircrafts
  </h2>
  <div
    class="aircraftsList"
  >
    <div
      class="aircraftContainer"
    >
      <div>
        GABCD
         
        A320
      </div>
      <div>
        0
        %
      </div>
    </div>
  </div>
</div>
`;
